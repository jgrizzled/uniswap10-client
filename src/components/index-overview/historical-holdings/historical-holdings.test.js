import HistoricalHoldings from '.';

const component = (
  <HistoricalHoldings holdingsByDate={mockData.holdingsByDate} />
);

// smoke test
it('renders without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<Providers>component</Providers>, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', async () => {
  const wrapper = Enzyme.mount(component, { wrappingComponent: Providers });
  expect(wrapper).toMatchSnapshot();
});
