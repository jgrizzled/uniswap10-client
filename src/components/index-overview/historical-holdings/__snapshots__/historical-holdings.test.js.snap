// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<HistoricalHoldings
  holdingsByDate={
    Array [
      Object {
        "MKR": 0.5,
        "ZRX": 0.5,
        "date": "2020-01-01",
      },
      Object {
        "MKR": 0.5,
        "ZRX": 0.5,
        "date": "2020-01-01",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-AxiKw hgsICA"
    >
      <styled.div>
        <div
          className="sc-AxjAm qfvsW"
        >
          <styled.h2>
            <h2
              className="sc-AxirZ jDYAnS"
            >
              Historical Holdings
            </h2>
          </styled.h2>
          <ResponsiveContainer
            aspect={1.3636363636363635}
            debounce={0}
            height="100%"
            width="100%"
          >
            <div
              className="recharts-responsive-container"
              id="undefined"
              style={
                Object {
                  "height": "100%",
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "width": "100%",
                }
              }
            >
              <AreaChart
                barCategoryGap="10%"
                barGap={4}
                data={
                  Array [
                    Object {
                      "MKR": 0.5,
                      "ZRX": 0.5,
                      "date": "2020-01-01",
                    },
                    Object {
                      "MKR": 0.5,
                      "ZRX": 0.5,
                      "date": "2020-01-01",
                    },
                  ]
                }
                height={0}
                layout="horizontal"
                margin={
                  Object {
                    "bottom": 5,
                    "left": 5,
                    "right": 5,
                    "top": 5,
                  }
                }
                reverseStackOrder={false}
                stackOffset="none"
                width={0}
              />
              <ResizeDetector
                handleHeight={true}
                handleWidth={true}
                nodeType="div"
                onResize={[Function]}
                querySelector={null}
                refreshRate={1000}
                skipOnMount={false}
                targetDomEl={null}
              >
                <ChildWrapper>
                  <div />
                </ChildWrapper>
              </ResizeDetector>
            </div>
          </ResponsiveContainer>
        </div>
      </styled.div>
    </div>
  </styled.div>
</HistoricalHoldings>
`;
